java/src/test/java/org/hyperledger/fabric/client/ChaincodeEventsTest.java

```diff
@@ -57,6 +74,18 @@ void afterEach() {
         mocker.close();
     }
 
+    void assertChaincodeRequest(org.hyperledger.fabric.protos.gateway.ChaincodeEventsRequest actual, ExpectedRequest expectedRequest){
+        assertThat(actual.getChannelId()).isEqualTo(expectedRequest.channelName);
+        assertThat(actual.getChaincodeId()).isEqualTo(expectedRequest.chaincodeName);
+        assertThat(actual.getStartPosition().getTypeCase()).isEqualTo(expectedRequest.typeCase);
+        if(expectedRequest.blockNumber.isPresent()) {
+            assertThat(actual.getStartPosition().getSpecified().getNumber()).isEqualTo(expectedRequest.blockNumber.get());
+        }
+        if(expectedRequest.transactionId.isPresent()) {
+            assertThat(actual.getAfterTransactionId()).isEqualTo(expectedRequest.transactionId.get());
+        }
+    }
```

Comment:
I would be tempted to remove the assertions of channel and chaincode name for the tests that are focusing on start position and checkpoint. If they are already covered adequately in earlier tests of different permutations where they could be missed, great. If not, we could just have a specific test for channel and chaincode name so we don't need to worry about them in other tests.

It might be simpler to break what's left up a bit to make it (much) simpler. I _think_ these methods save some copy & paste of code, and the method names make it clearer what is being asserted. See what you think.
```suggestion
    void assertStartPosition(final org.hyperledger.fabric.protos.gateway.ChaincodeEventsRequest actual, final long blockNumber, final String transactionId) {
        assertThat(actual.getStartPosition().getTypeCase()).isEqualTo(Ab.SeekPosition.TypeCase.SPECIFIED);
        assertThat(actual.getStartPosition().getSpecified().getNumber()).isEqualTo(blockNumber);
        assertThat(actual.getAfterTransactionId()).isEqualTo(transactionId);
    }

    void assertNextCommit(final org.hyperledger.fabric.protos.gateway.ChaincodeEventsRequest actual) {
        assertThat(actual.getStartPosition().getTypeCase()).isEqualTo(Ab.SeekPosition.TypeCase.NEXT_COMMIT);
    }
```